engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -463,8 +492,23 @@ private void publishTupleData(int portId, Object obj)
         wsClient.publish(recordingNameTopic, map);
       }
     } catch (Exception ex) {
-      logger.warn("Error publishing tuple data", ex);
+      if (shouldLog(ex)) {
+        logger.warn("Error publishing tuple data {}", obj, ex);
+      }
+    }
+  }
+
+  private boolean shouldLog(Exception ex)
+  {
+    // It is a regular exception not related to tuple serialization
+    if (!(ex instanceof RuntimeException)) {
```

Comment:
I don't think it is very reliable to use RuntimeException as an indicator to serialization problem. Can call to streamCodec be in it's own try/catch block?